sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py

```diff
@@ -487,6 +488,8 @@ def _populate_arm_paramaters(self, workspace: Workspace) -> Tuple[dict, dict, di
             )
             # if default resource group does not exist yet, create rg and log analytics
             if not rg_is_existing:
+                # if need to create default resource group, delete past default rg deployments in case close to 10
+                clean_up_rg_deployments_on_subscription(self._credentials, self._subscription_id)
```

Comment:
Are you doing this as a lot of customers run into this issue? Should we consider educating customers on how to solve this issue using a document? Let's make this an opt-in mode if we have to do it. E.g. add an argument to the function to control this behavior.

Reply:
I am doing this as it was noticed by some runners (using SDKv1 actually) using more than 10 different regions for workspaces in the same subscription so I figured I should fix it here as well. This is very unlikely to have affected many users since we reuse existing default resource group and log analytics on a per region basis this only happens when that does not already exist. It would not really make sense to expose to the user for controlling this behavior. This should be hidden from the user since this only happens in the case when the user does not specify (their own) existing app insights and we need to provision one for them and thus also need to deploy the default resource group. Resource group deployments have a low limit on the subscription level of only 10. It makes sense to clean up previous ones created by our service. Also NOTE that deployments on subscription level are cleaned up automatically anyways after some period of time by ARM and that deleting a deployment from the history doesn't affect any of the resources that were deployed.